libnd4j/include/ops/declarable/platform/cudnn/ctcloss.cu

```diff
@@ -72,8 +72,9 @@ namespace platforms {
         CTCLossDesc  ctcLossDesc;
         CudnnTensor probsDesc, gradsDesc(false);
         bool calcGrads = !grads.isEmpty();
-        ctcLossDesc.set( err, CUDNN_DATA_FLOAT, CUDNN_LOSS_NORMALIZATION_SOFTMAX, CUDNN_PROPAGATE_NAN);
-        probsDesc.set( err, cudnnDataType(probs.dataType()), probs.rankOf() , dims, strides);
+        auto cudnnType = cudnnDataType(probs.dataType());
```

Comment:
Could we still explain why we're converting the cudnn type? Why are we forced to do that? A simple link to the cudnn docs with the relevant section is fine. It could be something like: cudnn only supports xyz data types, users may try to pass in abc data types.

It's mainly because we have a trade off here of either throwing an error for it not being the exact type or automatically doing a cast. Whenever you automatically do something, that means it's ok to do so. Knowing why would be good for future reference.

Reply:
I do not convert from one type to another. that conversion is from our type to CUDNN equivalent type.
 I just allow certain types  in platform_check. with ctc loss I allow float32 adn int32. I will check it in the official documentation. but it could be also undocumented limitation

Reply:
Yeah just make sure to document that. What the conversion is for is important.